scripts/support/runner/intel_s1000.py

```diff
@@ -0,0 +1,138 @@
+
+    def __init__(self,
+                 board_dir, xt_ocd_dir,
+                 ocd_topology, ocd_jtag_instr, gdb_flash_file,
+                 elf_name, gdb,
+                 gdb_port=DEFAULT_XT_GDB_PORT, debug=False):
+        super(intel_s1000BinaryRunner, self).__init__(debug=debug)
+        self.board_dir = board_dir
+        self.xt_ocd_dir = xt_ocd_dir
+        self.ocd_topology = ocd_topology
+        self.ocd_jtag_instr = ocd_jtag_instr
+        self.gdb_flash_file = gdb_flash_file
+        self.elf_name = elf_name
+        self.gdb_cmd = gdb
+        self.gdb_port = gdb_port
+
+    @classmethod
+    def name(cls):
+        return 'intel_s1000'
+
+    @classmethod
+    def do_add_parser(cls, parser):
+        # Required
+
+        # Optional
+        parser.add_argument('--gdb-port', default=DEFAULT_XT_GDB_PORT,
+                            help='xt-gdb port, defaults to 20000')
+        parser.add_argument('--xt-ocd-dir', default='/opt/Tensilica/xocd-12.0.4/xt-ocd',
+                            help='ocd-dir, defaults to /opt/Tensilica/xocd-12.0.4/xt-ocd')
+        parser.add_argument('--ocd-topology', default='topology_dsp0_flyswatter2.xml',
+                            help='ocd-topology, defaults to topology_dsp0_flyswatter2.xml')
+        parser.add_argument('--ocd-jtag-instr', default='dsp0_gdb.txt',
+                            help='ocd-jtag-instr, defaults to dsp0_gdb.txt')
+        parser.add_argument('--gdb-flash-file', default='load_elf.txt',
+                            help='gdb-flash-file, defaults to load_elf.txt')
+
+    @classmethod
+    def create_from_args(command, args):
+        return intel_s1000BinaryRunner(
+            args.board_dir, args.xt_ocd_dir,
+            args.ocd_topology, args.ocd_jtag_instr, args.gdb_flash_file,
+            args.kernel_elf, args.gdb,
+            gdb_port=args.gdb_port, debug=args.verbose)
+
+    def do_run(self, command, **kwargs):
+        kwargs['ocd-topology'] =  path.join(self.board_dir, 'support',  self.ocd_topology)
+        kwargs['ocd-jtag-instr'] = path.join(self.board_dir, 'support', self.ocd_jtag_instr)
+        kwargs['gdb-flash-file'] = path.join(self.board_dir, 'support', self.gdb_flash_file)
+
+        if command == 'flash':
+            self.flash(**kwargs)
+        elif command == 'debugserver':
+            self.debugserver(**kwargs)
+        else:
+            self.do_debug()
+
+    def flash(self, **kwargs):
+        topology_file = kwargs['ocd-topology']
+        jtag_instr_file = kwargs['ocd-jtag-instr']
+        gdb_flash_file = kwargs['gdb-flash-file']
+
+        self.print_gdbserver_message(self.gdb_port)
+        server_cmd = [self.xt_ocd_dir,
+                        '-c', topology_file,
+                        '-I', jtag_instr_file]
+
+        # Start the server
+        # Note that XTOCD always fails the first time. It has to be
+        # relaunched the second time to work.
+        self.popen_ignore_int(server_cmd)
+        time.sleep(3)
```

Comment:
So this board needs to wait 6 seconds total for the command to fail and then be re-run before a flash can succeed? Sorry to hear it :(, sounds like silicon issues.

That's funny and unfortunate, and no problem for other runners, but I am not sure if popen_ignore_int is the right API to be used here. Do you really need to ignore SIGINT? Or is there a missing superclass call for creating a popen object that doesn't require a new process group?

Reply:
No idea if it is a silicon issue but this is how the board is being used. Please note there will be some verbose prints coming on the console, so 6 seconds are tolerable.

I am using popen_ignore_int so that i can move over to the next command. If i use check_call, it will be blocking. As you can see, multiple commands have to be entered and automation is not possible if they are blocking calls.